go/worker/common/committee/runtime_host.go

```diff
@@ -60,23 +63,35 @@ func (h *computeRuntimeHostHandler) Handle(ctx context.Context, body *protocol.B
 		span, sctx := opentracing.StartSpanFromContext(ctx, "storage.Sync")
 		defer span.Finish()
 
-		// Prioritize nodes that signed the last storage receipts.
-		h.node.CrossNode.Lock()
-		blk := h.node.CurrentBlock
-		h.node.CrossNode.Unlock()
-		if blk != nil {
-			sctx = storage.WithNodePriorityHintFromSignatures(sctx, blk.Header.StorageSignatures)
+		var rs syncer.ReadSyncer
+		switch rq.Endpoint {
+		case protocol.HostStorageEndpointRuntime:
+			// Runtime storage.
+			rs = h.storage
+
+			// Prioritize nodes that signed the last storage receipts.
+			h.node.CrossNode.Lock()
+			blk := h.node.CurrentBlock
+			h.node.CrossNode.Unlock()
```

Comment:
Was already wrong before, but this lock should be held until after line 78. Should probably just change tihs part to:

```
h.node.CrossNode.Lock()
if blk := h.node.CurrentBlock; blk != nil {
...
}
h.node.CrossNode.Unlock()
```

Reply:
Why would the lock need to be held there? The part after the unlock is only accessing `blk` (and not `h.node.CurrentBlock`) which has already been stored in a local variable and cannot change?

Reply:
Ah right, and also the `Block` itself won't ever change (other than be replaced with a different block) so indeed it's fine.